________________________________________________________________________________

# <h1 align="center"><font color='orange'>**CHI-SQUARE TEST**</font></h1>
________________________________________________________________________________

**Association between Device Type and Customer Satisfaction**

**Background:**

Mizzare Corporation has collected data on customer satisfaction levels for two types of smart home devices: Smart Thermostats and Smart Lights.
They want to determine if there's a significant association between the type of device purchased and the customer's satisfaction level.

**Data Provided:**

The data is summarized in a contingency table showing the counts of customers in each satisfaction level for both types of devices:

Satisfaction | Smart Thermostat | Smart Light |	Total
------------ | ---------------- | ----------- | -----
Very Satisfied |	50 |	70 |	120
Satisfied |	80 |	100 |	180
Neutral |	60 |	90 |	150
Unsatisfied |	30 |	50 |	80
Very Unsatisfied |	20 |	50 |	70
Total |	240 |	360 |	600


**Objective:**

To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

## **Steps to perform the test:**

**State the Hypotheses:**

Null Hypothesis: There is no assocation between type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

Alternate Hypothesis: There is assocation between type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

**Compute the Chi-Square Statistic**

alpha = 0.05

CI = 95%

$χ^2= \sum \frac {(O^i − E^i)^2}{E^i}$

where $O^i$ is the observed frequency and $E^i$ is the expected frequency



**Determine the degrees of freedom (df)**

$df = (r -1) * (c-1)$

where r is the number of rows and c is the number of columns.

**Compare the Chi-Square statistic** : to the critical value from the Chi-Square distribution table at a chosen significance level (usually 0.05).

Let's perform the calculations:

**Step 1: Calculate Expected Frequencies**

The expected frequency for each cell is calculated as:

$E_{ij} = \frac {(Row Total)  *  (Column Total)}{Grand Total}$

**Step 2: Compute the Chi-Square Statistic**

In [1]:
import numpy as np
from scipy.stats import chi2_contingency

# Observed frequencies
observed = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

# Perform the Chi-Square test
chi2, p, dof, expected = chi2_contingency(observed)

print("Chi-Square statistic (χ²):", chi2)
print("p-value:", p)
print("Degrees of freedom (df):", dof)
print("Expected frequencies:")
print(expected)

Chi-Square statistic (χ²): 5.638227513227513
p-value: 0.22784371130697179
Degrees of freedom (df): 4
Expected frequencies:
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


**Interpretation:**

The p-value is 0.228 which is greater than the significance value of 0.05. This means that we fail to reject the null hypothesis.

**Conclusion:**

There is no statistically significant association between the type of smart home device (Smart Thermostats vs. Smart Lights) and the customer satisfaction level. The distribution of satisfaction levels appears to be independent of the type of device purchased.